pkg/cloud/converters/eks_test.go

```diff
@@ -0,0 +1,195 @@
+					Value:  awsString("value1"),
+					Effect: awsString(eks.TaintEffectNoExecute),
+				},
+				{
+					Key:    awsString("key2"),
+					Value:  awsString("value2"),
+					Effect: awsString(eks.TaintEffectPreferNoSchedule),
+				},
+			},
+			expected: expinfrav1.Taints{
+				{
+					Key:    "key1",
+					Value:  "value1",
+					Effect: expinfrav1.TaintEffectNoExecute,
+				},
+				{
+					Key:    "key2",
+					Value:  "value2",
+					Effect: expinfrav1.TaintEffectPreferNoSchedule,
+				},
+			},
+			expectErr: false,
+		},
+		{
+			name: "Unknown taint effect",
+			input: []*eks.Taint{
+				{
+					Key:    awsString("key1"),
+					Value:  awsString("value1"),
+					Effect: awsString("UnknownEffect"),
+				},
+			},
+			expected:  nil,
+			expectErr: true,
+		},
+		{
+			name: "Nil taint value",
+			input: []*eks.Taint{
+				{
+					Key:    awsString("key1"),
+					Value:  nil,
+					Effect: awsString(eks.TaintEffectNoSchedule),
+				},
+			},
+			expected: expinfrav1.Taints{
+				{
+					Key:    "key1",
+					Value:  "",
+					Effect: expinfrav1.TaintEffectNoSchedule,
+				},
+			},
+			expectErr: false,
+		},
+		{
+			name:      "Empty input slice",
+			input:     []*eks.Taint{},
+			expected:  expinfrav1.Taints{},
+			expectErr: false,
+		},
+		{
+			name:      "Nil input",
+			input:     nil,
+			expected:  expinfrav1.Taints{},
+			expectErr: false,
+		},
+		{
+			name: "Mixed valid and invalid taints",
+			input: []*eks.Taint{
+				{
+					Key:    awsString("key1"),
+					Value:  awsString("value1"),
+					Effect: awsString(eks.TaintEffectNoExecute),
+				},
+				{
+					Key:    awsString("key2"),
+					Value:  awsString("value2"),
+					Effect: awsString("InvalidEffect"),
+				},
+			},
+			expected:  nil,
+			expectErr: true,
+		},
+		{
+			name: "Empty key",
+			input: []*eks.Taint{
+				{
+					Key:    awsString(""),
+					Value:  awsString("value1"),
+					Effect: awsString(eks.TaintEffectNoSchedule),
+				},
+			},
+			expected: expinfrav1.Taints{
+				{
+					Key:    "",
+					Value:  "value1",
+					Effect: expinfrav1.TaintEffectNoSchedule,
+				},
+			},
+			expectErr: false,
+		},
+		{
+			name: "Empty value",
+			input: []*eks.Taint{
+				{
+					Key:    awsString("key1"),
+					Value:  awsString(""),
+					Effect: awsString(eks.TaintEffectNoSchedule),
+				},
+			},
+			expected: expinfrav1.Taints{
+				{
+					Key:    "key1",
+					Value:  "",
+					Effect: expinfrav1.TaintEffectNoSchedule,
+				},
+			},
+			expectErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := TaintsFromSDK(tt.input)
+			if tt.expectErr {
+				assert.Error(t, err)
+				assert.Nil(t, result, "Expected result to be nil on error")
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, result, "Converted taints do not match expected")
+			}
+		})
+	}
+}
+
+// Helper function to create AWS String pointers.
+func awsString(value string) *string {
+	return &value
+}
```

Comment:
We already use `ptr.To[string](...)` for this (or `aws.String(...)` in older code)

(minor, rest looks good)